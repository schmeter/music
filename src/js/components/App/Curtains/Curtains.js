import React from 'react';
import classNames from 'classnames';
import autobind from 'autobind-decorator';

class Curtains extends React.Component {
    state = {
        curtainsOpen: true
    };

    @autobind
    handleClickCurtains(e) {
        e.preventDefault();
        this.setState({ curtainsOpen: !this.state.curtainsOpen });
    }

    render() {
        return (
            <>
                {['left', 'right'].map(direction => (
                    <div
                        key={direction}
                        className={classNames(
                            'curtains',
                            direction,
                            { open: this.state.curtainsOpen }
                        )}
                        onClick={this.handleClickCurtains}
                    />
                ))}
            </>
        );
    }
}

export default Curtains;
