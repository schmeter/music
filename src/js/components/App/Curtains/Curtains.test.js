import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Curtains from './Curtains';

const mockedEvent = {
    preventDefault: jest.fn()
};

it('renders correctly', () => {
    const props = {
        path: 'http://test',
        onClickFile: jest.fn()
    };
    const component = shallow(<Curtains {...props} />);

    component.find('div').first().simulate('click', mockedEvent);

    expect(toJson(component)).toMatchSnapshot();
    expect(toJson(component)).toMatchSnapshot();
});
