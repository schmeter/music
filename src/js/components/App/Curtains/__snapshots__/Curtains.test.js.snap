// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <div
    className="curtains left"
    key="left"
    onClick={[Function]}
  />
  <div
    className="curtains right"
    key="right"
    onClick={[Function]}
  />
</Fragment>
`;

exports[`renders correctly 2`] = `
<Fragment>
  <div
    className="curtains left"
    key="left"
    onClick={[Function]}
  />
  <div
    className="curtains right"
    key="right"
    onClick={[Function]}
  />
</Fragment>
`;
